src/coreclr/tools/Common/Internal/Runtime/ReadyToRunConstants.cs

```diff
@@ -233,6 +233,7 @@ public enum ReadyToRunHelper
         WriteBarrier                = 0x30,
         CheckedWriteBarrier         = 0x31,
         ByRefWriteBarrier           = 0x32,
+        AssignStruct                = 0x33,
```

Comment:
This is a bit misleading name. Unlike the original implementation, the current contract of the helper does not allow assignment of arbitrary structs. It can be only used to assign structs (or other memory ranges) that contain GC references and their size is pointer size aligned.

The name should end with `WriteBarrier` to communicate that the helper is only meant to be used for blocks with GC references. `BulkWriteBarrier`, `MemMoveWriteBarrier`, `BulkMoveWithWriteBarrier`, or something along these lines.

Reply:
Renamed to BulkWriteBarrier